File: ios/Modules/LoggerModule.m
//
//  LoggerModule.m
//  ReactNativeTrace
//
//  Created by Alejo Castaño on 10/11/24.
//
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(LoggerModule, NSObject)

RCT_EXTERN_METHOD(initialize:(NSString *)apiKey enableDebug:(BOOL)enableDebug)
RCT_EXTERN_METHOD(log:(NSString *)level message:(NSString *)message metadata:(NSDictionary *)metadata)

@end

File: ios/Modules/LoggerModule.swift
//
//  LoggerModule.swift
//  ReactNativeTrace
//
//  Created by Alejo Castaño on 10/11/24.
//
import React

@objc(LoggerModule)
class LoggerModule: NSObject {
  
  @objc
  func initialize(_ apiKey: String, enableDebug: Bool) {
    print("Logger Initialized with API Key: \(apiKey)")
    print("enableDebug status: \(enableDebug)")
  }
  
  @objc
  func log(_ level: String, message: String, metadata: [String: Any]) {
    print("[\(level.uppercased())] \(message): \(metadata)")
  }
}
